WooCommerce/Classes/ViewRelated/Filters/FilterHistoryView.swift

```diff
@@ -0,0 +1,173 @@
+        "filterHistoryView.emptyState",
+        value: "No past filters found",
+        comment: "Label on the empty state of the Filter History view"
+    )
+
+    private let recent = NSLocalizedString(
+        "filterHistoryView.recent",
+        value: "Recent",
+        comment: "Header label of the Filter History view"
+    )
+
+    private let clearHistory = NSLocalizedString(
+        "filterHistoryView.clearHistory",
+        value: "Clear History",
+        comment: "Button to clear all history on the Filter History view"
+    )
+
+    private let clearHistoryConfirmation = NSLocalizedString(
+        "filterHistoryView.clearHistoryConfirmation",
+        value: "Are you sure you want to clear all the filter history?",
+        comment: "Message to confirm clearing all history on the Filter History view"
+    )
+
+    init(viewModel: ViewModel, onSelection: @escaping (ViewModel.Criteria) -> Void) {
+        self.viewModel = viewModel
+        self.onSelection = onSelection
+    }
+
+    var body: some View {
+        NavigationStack {
+            VStack {
+                if error != nil || savedFilters.isEmpty {
+                    emptyStateView
+                } else {
+                    filterListView
+                }
+            }
+            .navigationTitle(title)
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .cancellationAction) {
+                    Button(cancel) {
+                        onDismiss()
+                    }
+                }
+                ToolbarItem(placement: .confirmationAction) {
+                    Button(apply) {
+                        guard let selectedFilter else { return }
+                        onSelection(selectedFilter)
+                        onDismiss()
+                    }
+                    .disabled(selectedFilter == nil)
+                }
+            }
+        }
+        .task {
+            do {
+                savedFilters = try await viewModel.retrieveFilterHistory()
+            } catch {
+                self.error = error
+            }
+        }
+        .alert(clearHistoryConfirmation, isPresented: $shouldConfirmClearingHistory) {
+            Button(cancel, role: .cancel) {}
+            Button(clearHistory, role: .destructive) {
+                viewModel.clearAllFilterHistory()
+                savedFilters = []
+                selectedFilter = nil
+            }
+        }
+    }
+}
+
+private extension FilterHistoryView {
+    var filterListView: some View {
+        List {
+            Section {
+                ForEach(savedFilters, id: \.readableString) { filter in
+                    SelectableItemRow(title: filter.readableString,
+                                      selected: selectedFilter == filter,
+                                      displayMode: .compact,
+                                      alignment: .trailing)
+                    .onTapGesture {
```

Comment:
Opinion: Should we use `.listRowInsets(.zero)` to get rid of the extra padding? 

<img src="https://github.com/user-attachments/assets/60ac116c-73e4-4ece-b06f-4ec35ec25e57" width="320"/>

Reply:
Good catch - I added this in a later PR (#14831) so I'll keep this for now. Thanks!